### DISCUSSION TITLE
Oauth 2 token question

### DISCUSSION THREAD

**Discussion by IrfanZ0** [community/community#58282] [New to GitHub] [answered]
Hi, I'm new not to Github but to performing Oauth 2 token authorization for an Android application that I'm trying to develop.  In my repository on Github, I have a JSON file that my app needs to perform retrofit functions like GET and POST.  If a user downloads my application from Play Store and tries to retrieve or save their data, do these users need to sign up for a GitHub account and log in?  In other words, do I need to go through the OAuth route for those users to access their information from GitHub?  If I'm not mistaken, I believe OAUTH is for developers trying to access a repository, correct?  On a related note, I'm hearing that Github is removing the username and password for the OAuth token authorization, is this true?

Thanks for any clarification

**Comment by the-officialjosh** [comment, marked as answer]
Hello,

In order for users to access their data from your Android application that interacts with your GitHub repository, they do not necessarily need to sign up for a GitHub account or go through the OAuth route. The OAuth flow is typically used when you, as a developer, want to access specific resources or perform actions on behalf of a user on GitHub.

If your application is fetching data from a JSON file stored in your GitHub repository, you can make the repository and file publicly accessible. This way, users of your Android application can retrieve the data without requiring a GitHub account or authentication.

Regarding the removal of username and password for OAuth token authorization, it's true that GitHub is deprecating the use of username and password for authentication in favor of personal access tokens (PATs) or OAuth tokens. PATs or OAuth tokens provide a more secure and granular way to authenticate and authorize API requests. It's recommended to update your authentication flow to use tokens instead of username and password.

I hope this clarifies your questions. Let me know if you need further assistance!

**Reply by IrfanZ0** [reply]
Hey the-officialjosh, thanks for the clarification.  It makes a lot of sense now.  I have one more question in regard to making available my JSON file.  I have made both my repository and JSON file public and placed them at the following location: https://github.com/IrfanZ0/ShoppingWishList/blob/main/shoppingwishlist/app/src/main/java/com/tritongames/shopwishlist/docs/contacts
I've tried to use Github Pages to set up my JSON file, but I was unsuccessful.  Is there a way to shorten this long URL?  I've seen some example URLs and they were pretty short like: https://github.com/docs or https://github.com/posts.  The reason is that my GET method for example is like this:
`package com.tritongames.shopwishlist.data.models

import retrofit2.Response
import retrofit2.http.GET

interface ShoppingCategoriesInterface {

    @GET("/IrfanZ0/Shopping-Wish-List/blob/master/ShoppingWishList/src/com/shop/shoppingwishlist/ShopCategories")
    suspend fun getAllCategoriesIcons() : Response<List<ShoppingCategoriesDataClass>>

}`

It would make it easier and cleaner for POST methods as well.

**Reply by the-officialjosh** [reply]
To have a shorter URL for your JSON file, you can utilize GitHub's "Raw" feature. Here's how you can do it:

1. Open your repository on GitHub.
2. Navigate to the location of your JSON file: `shoppingwishlist/app/src/main/java/com/tritongames/shopwishlist/docs/contacts`.
3. Click on the JSON file to open it.
4. At the top-right corner of the file view, you should see a button labeled "Raw." Click on it.
5. This will open the raw version of your JSON file in the browser. Copy the URL from the address bar.

By using the "Raw" URL, you can have a shorter and cleaner link to your JSON file. You can then update your GET method to use the Raw URL instead of the long GitHub URL.

Please note that the "Raw" URL may not have the same structure as the examples you mentioned (`https://github.com/docs` or `https://github.com/posts`). Those examples could be specific to certain repositories or organizations. However, by using the "Raw" URL, you can achieve a shorter and more direct link to your JSON file.

And please mark the other response as the answer to that question so it helps others with similar issues,
